src/modules/wallet/components/connectButton/index.tsx

```diff
@@ -14,3 +14,58 @@ export const ConnectButton: React.FC<ConnectButtonProps> = ({ handleClick }) =>
     </button>
   );
 };
+
+interface LaceButtonProps {
+  handleClick: () => void;
+}
+export const LaceButton: React.FC<LaceButtonProps> = ({ handleClick }) => {
+  const cardano = window?.cardano;
+  console.log('window.cardano: ' + JSON.stringify(Object.getOwnPropertyNames(cardano)));
+  // this is according to CIP-30
+  const lace = window?.cardano?.lace;
+  if (typeof lace === 'undefined') {
```

Comment:
why you check with `typeof`
when you have nothing in a condition clause, you can do it in opposite condition
`if(lace) then your codes...`

Reply:
It's so that we don't wrap the rest of the function in an if statement